contracts/modules/TransferManager/SingleTradeVolumeRestrictionManager.sol

```diff
@@ -0,0 +1,111 @@
+pragma solidity ^0.4.24;
+import "./ITransferManager.sol";
+
+/////////////////////
+// Module permissions
+/////////////////////
+//                           Owner       ADMIN
+// changeGlobalLimit           X           X
+// addExemptWallet             X           X
+// removeExemptWallet          X           X
+// setTransferLimitForWallet   X           X
+// removeTransferLimitForWallet X          X
+
+contract SingleTradeVolumeRestrictionManager is ITransferManager {
+  using SafeMath for uint256;
+
+  bool public globalTransferLimitInPercentage;
+
+  uint256 public globalTransferLimit;
+
+  mapping(address=>bool) public exemptWallets;
+
+  mapping(address => bool) public specialTransferLimitWallets;
+
+  mapping(address => uint) public specialTransferLimits;
+
+  bytes32 constant public ADMIN = "ADMIN";
+
+  constructor(address _securityToken, address _polyAddress) public
+  IModule(_securityToken, _polyAddress)
+  {
+
+  }
+
+  function verifyTransfer(address _from, address _to, uint256 _amount, bool /* _isTransfer */) public returns(Result) {
+    bool validTransfer;
+
+    if(paused) {
+      return Result.NA;
+    }
+
+    if(exemptWallets[_from]) return Result.NA;
+
+    if(specialTransferLimitWallets[_from]) {
+      if(globalTransferLimitInPercentage) {
+        validTransfer = (_amount.mul(100).div(ISecurityToken(securityToken).totalSupply())) <= specialTransferLimits[_from];
```

Comment:
Error in the calculation. You are giving the margin of 0.9 %. Suppose if the left size have the 15.7% and right side have the 15 % value then it still gets passed instead of failing.

Reply:
@satyamakgec I don't fully understand your comment.  Is your comment because of this warning from the solidity docs ? ```Division on integer literals used to truncate in earlier versions, but it will now convert into a rational number, i.e. 5 / 2 is not equal to 2, but to 2.5.```

Reply:
Also From PercentageTransferManager
```if(newBalance.mul(10**uint256(ISecurityToken(securityToken).decimals())).div(ISecurityToken(securityToken).totalSupply()) > maxHolderPercentage) {
                return Result.INVALID;
}```

Shouldn't newBalance be multipled by 10**18 explicitly instead of 10**decimals as maxHolderPercentage is multipled by (10**18/100) ?

Reply:
I don't think so because by default securitytoken have 18 decimals so in case of `PercentageTransferManager` it always give you the value right percentage calculation.
      
       newBalance = x * 10^d
       totalSupply = y * 10^d
       maxHolderPercentage = z * 10^d
       so it looks like this -  (x* 10^d * 10^d) / y* 10^d > (z * 10^d)/100      ... where d = 18 
       After calculation you can easily compare ((x * 10^18)/y) > (z* 10^16)

But in your case you are doing this 
     
       amount  = a* 10^d
       totalSupply = b* 10^d
       Limit= c* 10^d  (case 1)
       so it will look like this -  (a* 10^d * 100)/(b* 10^d) <= c* 10^d   .... where d = 18
       so your left hand side is always less than the right hand side.
       
       Limit = c (case 2)
       in this condition  - (155000 * 10^d * 100) / (1000000 * 10^d) <= 15   .... a= 155000, b = 1000000, c= 15 and d =18.
       After calculation it should be  15.5 <= 15  (and it will returns true as per my test on remix with 0.4.24 and 0.4.25 solidity version).

      and using case 2 you also can't cover the limits in decimal

Reply:
@satyamakgec Thanks for the reply. The first case, which I haven't implemented is missing a multiplication by 10^18. it should be ```(a* 10^d * 10^18)/(b* 10^d) <= c* 10^16```. I will implement this formula/calculation.

In the `PercentageTransferManager` -  the math should be (x* 10^d * 10^18) / y* 10^d > (z * 10^18)/100. The multiplication should be explicitly done by 10^18. I think this would work for security token of any decimals, not just 18